python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -190,20 +190,14 @@ class NearestNeighbors(Base):
 
         # Only need to store index if fit() was called
         if self.n_indices == 1:
-            state['X_m'] = cudf.DataFrame.from_gpu_matrix(self.X_m)
```

Comment:
`__getstate__` and `__setstate__` should be able to be removed now that these are provided in `cuml.Base`.